src/Core/src/Handlers/Switch/SwitchHandler.Android.cs

```diff
@@ -80,6 +80,12 @@ class CheckedChangeListener : Java.Lang.Object, CompoundButton.IOnCheckedChangeL
 			void CompoundButton.IOnCheckedChangeListener.OnCheckedChanged(CompoundButton? buttonView, bool isToggled)
 			{
 				Handler?.OnCheckedChanged(isToggled);
+
+				// Schedule shadow update after a short delay to allow toggle animation to finish
+				if (Handler?.VirtualView?.Shadow is not null)
+				{
+					Handler.PlatformView?.PostDelayed(() => MapShadow(Handler, Handler.VirtualView), 150);
```

Comment:
This time-based approach could be fragile, the 150ms delay may not align with actual animation duration in some device/API Level.

Reply:
Maybe using an AnimationListener and when the ended is a better approach. If not possible, could extract the delay in a constant? also, add a comment explaining it.